flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlot.java

```diff
@@ -305,4 +321,23 @@ public String toString() {
 		return "TaskSlot(index:" + index + ", state:" + state + ", resource profile: " + resourceProfile +
 			", allocationId: " + (allocationId != null ? allocationId.toString() : "none") + ", jobId: " + (jobId != null ? jobId.toString() : "none") + ')';
 	}
+
+	@Override
+	public void close() {
+		verifyMemoryFreed();
```

Comment:
are there valid use-cases where this assumption is violated? If not, how about adding an assert in `verifyMemoryFreed`?

Reply:
I cannot come up with any valid use-cases for this check to fail and firstly I had an assertion. I had some failing test cases previously (probably for some new blink code) but CI does not fail anymore, they might have already been fixed. I tried CI with a commit inserting the assertion: https://travis-ci.org/azagrebin/flink/builds/613505573. It is also green.

Anyways, as discussed offline, memory leak after finishing task should not fail the whole TM but we have to provide a warning.